### PR TITLE
Migrated samples to WM b2. migrated to view binding.

### PR DESCRIPTION
**Pull request by soalb-m**
Migrated samples to WM b2. 
Migrated to view binding.
Bugfixes

### CODE REVIEW

File: DualView/src/main/java/com/microsoft/device/display/samples/dualview/DualViewActivity.kt [hunk 1]
```diff
@@ -42,18 +54,19 @@ class DualViewActivity : AppCompatActivity(), ScreenInfoListener {
...
+                windowInfoRepository.windowLayoutInfo.collect { info ->
+                    onScreenInfoChanged(info)
+                }
+            }
+        }
     }
 
-    override fun onScreenInfoChanged(screenInfo: ScreenInfo) {
-        if (screenInfo.isDualMode()) {
+    private fun onScreenInfoChanged(windowLayoutInfo: WindowLayoutInfo) {
```
Comment (bimiron) [inline DualView/src/main/java/com/microsoft/device/display/samples/dualview/DualViewActivity.kt:68]: Maybe also rename this method?

File: HingeAngle/src/main/java/com/microsoft/device/display/samples/hingeangle/HingeAngleActivity.kt [hunk 2]
```diff
@@ -34,22 +46,23 @@ class HingeAngleActivity : AppCompatActivity(), ScreenInfoListener {
...
+                windowInfoRepository.windowLayoutInfo.collect { info ->
+                    onScreenInfoChanged(info)
+                }
+            }
+        }
     }
 
-    override fun onScreenInfoChanged(screenInfo: ScreenInfo) {
-        addFragments(screenInfo)
+    private fun onScreenInfoChanged(windowLayoutInfo: WindowLayoutInfo) {
```
Comment (bimiron) [inline HingeAngle/src/main/java/com/microsoft/device/display/samples/hingeangle/HingeAngleActivity.kt:60]: Same here - maybe also rename this method?

File: SurfaceDuoSamples/build.gradle [hunk 3]
```diff
@@ -4,10 +4,8 @@
  *
  */
 
-plugins {
-    id 'com.android.application'
-    id 'kotlin-android'
-}
+apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
```
Comment (bimiron) [inline SurfaceDuoSamples/build.gradle:8]: Why was the gradle plugins syntax changed?

File: dependencies.gradle [hunk 4]
```diff
@@ -5,12 +5,12 @@
...
-    kotlinVersion = "1.4.30"
-    compileSdkVersion = 29
-    buildToolsVersion = '29.0.2'
+    gradlePluginVersion = "7.0.2"
+    kotlinVersion = "1.5.0"
+    compileSdkVersion = 31
+    buildToolsVersion = '30.0.3'
     targetSdkVersion = compileSdkVersion
-    minSdkVersion = compileSdkVersion
+    minSdkVersion = 25
```
Comment (bimiron) [inline dependencies.gradle:13]: Which components we use require this minSdkVersion? Is it not possible to have it in sync with the SDK, which has minSdkVersion = 21?

File: TwoPage/src/main/java/com/microsoft/device/display/samples/twopage/TwoPageActivity.kt [hunk 5]
```diff
@@ -48,25 +59,28 @@ class TwoPageActivity : AppCompatActivity(), OnPageChangeListener, ScreenInfoLis
...
+                        )
+                }
+            }
+        }
     }
 
-    override fun onConfigurationChanged(newConfig: Configuration) {
-        super.onConfigurationChanged(newConfig)
-        ScreenManagerProvider.getScreenManager().onConfigurationChanged()
+    private fun onScreenInfoChanged(windowLayoutInfo: WindowLayoutInfo) {
```
Comment (bimiron) [inline TwoPage/src/main/java/com/microsoft/device/display/samples/twopage/TwoPageActivity.kt:78]: Same here - maybe rename this method?

File: MultipleInstances/src/main/java/com/microsoft/device/display/samples/multipleinstances/MultipleInstancesBaseActivity.kt [hunk 6]
```diff
@@ -36,21 +47,25 @@ abstract class MultipleInstancesBaseActivity : AppCompatActivity(), ScreenInfoLi
...
 
-    override fun onScreenInfoChanged(screenInfo: ScreenInfo) {
-        first_screen_text.text = getFirstScreenText()
+    private fun onScreenInfoChanged(windowLayoutInfo: WindowLayoutInfo) {
+        findViewById<AppCompatTextView>(R.id.first_screen_text).text = getFirstScreenText()
 
-        if (screenInfo.isDualMode()) {
-            second_screen_text.text = getSecondScreenText()
+        if (windowLayoutInfo.isInDualMode()) {
+            findViewById<AppCompatTextView>(R.id.second_screen_text).text = getSecondScreenText()
```
Comment (CesarValiente) [inline MultipleInstances/src/main/java/com/microsoft/device/display/samples/multipleinstances/MultipleInstancesBaseActivity.kt:68]: same here (re: viewbinding usage)

File: MultipleInstances/src/main/java/com/microsoft/device/display/samples/multipleinstances/MultipleInstancesBaseActivity.kt [hunk 7]
```diff
@@ -36,21 +47,25 @@ abstract class MultipleInstancesBaseActivity : AppCompatActivity(), ScreenInfoLi
...
+                    )
+                }
+            }
+        }
     }
 
-    override fun onScreenInfoChanged(screenInfo: ScreenInfo) {
-        first_screen_text.text = getFirstScreenText()
+    private fun onScreenInfoChanged(windowLayoutInfo: WindowLayoutInfo) {
+        findViewById<AppCompatTextView>(R.id.first_screen_text).text = getFirstScreenText()
```
[the remark this answers is not in the document]
Comment (soalb-m) [reply MultipleInstances/src/main/java/com/microsoft/device/display/samples/multipleinstances/MultipleInstancesBaseActivity.kt:65]: first_screen_text is a view from a layout inflated by the FoldableLayout.
It is not visible as a child view for the binding object.

File: TwoPage/src/main/java/com/microsoft/device/display/samples/twopage/TwoPageActivity.kt [hunk 8]
```diff
@@ -48,25 +59,28 @@ class TwoPageActivity : AppCompatActivity(), OnPageChangeListener, ScreenInfoLis
...
-
-    override fun onPause() {
-        super.onPause()
-        ScreenManagerProvider.getScreenManager().removeScreenInfoListener(this)
+    private fun initWindowLayoutInfo() {
+        val windowInfoRepository = windowInfoRepository()
+        lifecycleScope.launch(Dispatchers.Main) {
+            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                windowInfoRepository.windowLayoutInfo.collect { info ->
+                    // Temporary fix for FoldableLayout delay bug
```
[the remark this answers is not in the document]
Comment (soalb-m) [reply TwoPage/src/main/java/com/microsoft/device/display/samples/twopage/TwoPageActivity.kt:67]: Both the Activity and the FoldableLyaout register callbacks to get the WindowLayoutInfo.
The Activity gets called before the FoldableLayout and tries to use views that were not yet inflated.

File: DualView/src/main/java/com/microsoft/device/display/samples/dualview/DualViewActivity.kt (hunk 1 above)
Comment (soalb-m) [reply DualView/src/main/java/com/microsoft/device/display/samples/dualview/DualViewActivity.kt:68]: Fixed

File: HingeAngle/src/main/java/com/microsoft/device/display/samples/hingeangle/HingeAngleActivity.kt (hunk 2 above)
Comment (soalb-m) [reply HingeAngle/src/main/java/com/microsoft/device/display/samples/hingeangle/HingeAngleActivity.kt:60]: Fixed

File: SurfaceDuoSamples/build.gradle (hunk 3 above)
Comment (soalb-m) [reply SurfaceDuoSamples/build.gradle:8]: Align with the way we used it in the SDK

File: dependencies.gradle (hunk 4 above)
Comment (soalb-m) [reply dependencies.gradle:13]: In the PenEventDrawer - context.getColor(colorResId) requires lev 23
In the DragSourceFragment - view.startDragAndDrop() requires lev 24
In the Shortcuts.kt - ShortcutManager requires lev 25

File: TwoPage/src/main/java/com/microsoft/device/display/samples/twopage/TwoPageActivity.kt (hunk 5 above)
Comment (soalb-m) [reply TwoPage/src/main/java/com/microsoft/device/display/samples/twopage/TwoPageActivity.kt:78]: Fixed

File: MultipleInstances/src/main/java/com/microsoft/device/display/samples/multipleinstances/MultipleInstancesBaseActivity.kt (hunk 6 above)
Comment (soalb-m) [reply MultipleInstances/src/main/java/com/microsoft/device/display/samples/multipleinstances/MultipleInstancesBaseActivity.kt:68]: Same as above: the view is not a direct child of the binding. 
It is added later by the FoldableLayout

File: MultipleInstances/src/main/java/com/microsoft/device/display/samples/multipleinstances/MultipleInstancesBaseActivity.kt [hunk 9]
```diff
@@ -36,21 +47,25 @@ abstract class MultipleInstancesBaseActivity : AppCompatActivity(), ScreenInfoLi
...
-
-    override fun onPause() {
-        super.onPause()
-        ScreenManagerProvider.getScreenManager().removeScreenInfoListener(this)
+    private fun initWindowLayoutInfo() {
+        val windowInfoRepository = windowInfoRepository()
+        lifecycleScope.launch(Dispatchers.Main) {
+            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                windowInfoRepository.windowLayoutInfo.collect { info ->
+                    binding.foldableLayout.postDelayed(
```
[the remark this answers is not in the document]
Comment (soalb-m) [reply MultipleInstances/src/main/java/com/microsoft/device/display/samples/multipleinstances/MultipleInstancesBaseActivity.kt:55]: Both the Activity and the FoldableLyaout register callbacks to get the WindowLayoutInfo.
The Activity gets called before the FoldableLayout and tries to use views that were not yet inflated.
The rotation works.

File: .editorconfig [hunk 10]
```diff
@@ -3,4 +3,4 @@
 # Comma-separated list of rules to disable (Since 0.34.0)
 # Note that rules in any ruleset other than the standard ruleset will need to be prefixed
 # by the ruleset identifier.
-disabled_rules = final-newline
+disabled_rules = final-newline,  import-ordering
```
[the remark this answers is not in the document]
Comment (soalb-m) [reply .editorconfig:6]: OK, 
The issue is that the ktlint fails without this rule.
I tried a few possible solutions, for example the one pasted bellow, but only this works 

https://github.com/pinterest/ktlint/issues/1228

File: PenEvents/src/main/java/com/microsoft/device/display/samples/pen/PenEventsActivity.kt [hunk 11]
```diff
@@ -40,19 +44,29 @@ class PenEventsActivity : AppCompatActivity() {
...
     }
 
     private fun displayValues(event: MotionEvent) {
-        device_info.text = parseToolType(event.getToolType(0))
-        pressure_value.text = getString(R.string.pen_events_pressure_value, event.pressure.toString())
-        orientation_value.text = getString(R.string.pen_events_orientation_value, event.orientation.toString())
-        button_state_value.text = getString(R.string.pen_events_button_state_value, event.buttonState)
-        location_value.text = getString(R.string.pen_events_location_value, event.rawX.toString(), event.rawY.toString())
-        tilt_value.text = getString(R.string.pen_events_tilt_value, event.getAxisValue(MotionEvent.AXIS_TILT).toString())
+        binding.deviceInfo.text = parseToolType(event.getToolType(0))
```
[the remark this answers is not in the document]
Comment (soalb-m) [reply PenEvents/src/main/java/com/microsoft/device/display/samples/pen/PenEventsActivity.kt:55]: Fixed

**Review by ancirja-m — APPROVED** [review]

**Review by bimiron — APPROVED** [review]

**Pull request merged**